config/config.go

```diff
@@ -86,6 +90,8 @@ const (
 	SigP256sm2 = "p256sm2"
 )
 
+var chainID sync.Once
```

Comment:
not sure if this is necessary since you only assign the value once

Reply:
this is for safety, that multiple calls to SetExternChainID() will only get called once